// api-devup/dist/cjs/lib/API/utils/main.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const main_1 = require("./../main");
const md5Generate = __importStar(require("./methods/md5Generate"));
const getServerTime = __importStar(require("./methods/getServerTime"));
const createShortLink = __importStar(require("./methods/createShortLink"));
const notificationsLinks = __importStar(require("./methods/notificationsLinks"));
const getWebInfo = __importStar(require("./methods/getWebInfo"));
class Utils extends main_1.Builder {
    constructor(params) {
        super(params);
        this.md5Generate = (params) => __awaiter(this, void 0, void 0, function* () { return this.postRequest(md5Generate.method, params); });
        this.getServerTime = (params) => __awaiter(this, void 0, void 0, function* () { return this.postRequest(getServerTime.method, params); });
        this.createShortLink = (params) => __awaiter(this, void 0, void 0, function* () { return this.postRequest(createShortLink.method, params); });
        this.notificationsLinks = (params) => __awaiter(this, void 0, void 0, function* () { return this.postRequest(notificationsLinks.method, params); });
        this.getWebInfo = (params) => __awaiter(this, void 0, void 0, function* () { return this.postRequest(getWebInfo.method, params); });
    }
}
exports.default = Utils;
