// api-devup/dist/cjs/core.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DDOS_Guard = exports.DevUpError = exports.url = exports.version = void 0;
exports.version = "1.3.5";
exports.url = "https://api.dev-up.ru/method/";
class ModuleError extends Error {
    constructor(message) {
        super(message);
        Object.setPrototypeOf(this, ModuleError.prototype);
    }
    get [Symbol.toStringTag]() {
        return this.constructor.name;
    }
    /**
     * Возвращает содержимое ошибки в JSON
     */
    toJSON() {
        const json = {};
        for (const key of Object.getOwnPropertyNames(this)) {
            json[key] = this[key];
        }
        return json;
    }
}
/**
 * Основной класс ошибок которые могут произойти в модуле
 * @hideconstructor
 */
class DevUpError extends ModuleError {
    constructor({ err_code: code, err_critical_lvl: level, err_msg: message, params: params, }) {
        super(message);
        this.code = code;
        this.level = level;
        this.params = params;
        this.name = this.constructor.name;
        Object.setPrototypeOf(this, DevUpError.prototype);
    }
}
exports.DevUpError = DevUpError;
class DDOS_Guard extends ModuleError {
    constructor() {
        super("The number of requests per second has been exceeded, try again later");
        Object.setPrototypeOf(this, DDOS_Guard.prototype);
    }
}
exports.DDOS_Guard = DDOS_Guard;
